mirage/utils/utils.py

```diff
@@ -153,6 +153,47 @@ def calc_frame_time(instrument, aperture, xdim, ydim, amps):
     return ((1.0 * xs / amps + colpad) * (ys + rowpad) + fullpad) * 1.e-5
 
 
+def check_niriss_filter(filter,pupil):
+    """
+    This is a utility function that checks the FILTER and PUPIL parameters read in from the .yaml file and makes sure
+    that for NIRISS the filter and pupil names are correct, releaving the user of the need to remember which of the 12 
+    filters are in the filter wheel and which are in the pupil wheel.  If the user puts the correct values for filter and 
+    pupil nothing is done, but the user can just put the filter name in the filter parameter and CLEAR in the pupil 
+    parameter, in which case this routine sorts out which value actually goes where.  Hence for example one can have a 
+    parameter file with FILTER = F277W and PUPIL = CLEAR or FILTER = F090W and PUPIL = CLEAR for imaging and either will 
+    produce the desired result although the actual pairings would be F277W/CLEARP and CLEAR/F090W respectively.
+    
+    The code also corrects CLEARP to CLEAR if needed.
+    
+    Parameters:
+    
+    filter:   a string variable, assumed to be the self.params['Readout']['filter'] value from the .yaml file
```

Comment:
Not super-important, but 'filter' is actually a built-in function in python. https://www.w3schools.com/python/ref_func_filter.asp
I try to avoid using 'filter' as a variable name, in favor of 'filter_name' or 'filter_value'. It does still work with 'filter', but it can be a little confusing to people reading the code.